### PR TITLE
Jzhang/bugfix/tup 20511 need to limit the number of entry on the custom uri mapping json

### PR DESCRIPTION
**Pull request by jiezhang-tlnd**
https://jira.talendforge.org/browse/TUP-20511

### CODE REVIEW

File: main/plugins/org.talend.librariesmanager.ui/src/main/java/org/talend/librariesmanager/ui/actions/ImportExternalJarAction.java [hunk 1]
```diff
@@ -137,7 +137,9 @@ private boolean installFile(Set<String> modulesNeededNames, File file, String mo
         LibManagerUiPlugin.getDefault().getLibrariesService().deployLibrary(file.toURL(), mvnURI, false);
         if (!modulesNeededNames.contains(moduleName)) {
             String mavenUri = MavenUrlHelper.generateMvnUrlForJarName(moduleName, true, true);
-            CustomUriManager.getInstance().put(mavenUri, mavenUri);
+//            if(ModulesNeededProvider.getAllModuleNamesFromIndex().contains(moduleName)) {
+//            	CustomUriManager.getInstance().put(mavenUri, mavenUri);
+//            }
```
Comment (nrousseau) [inline main/plugins/org.talend.librariesmanager.ui/src/main/java/org/talend/librariesmanager/ui/actions/ImportExternalJarAction.java]: why keep a comment for a code unused?
To be removed here.

File: main/plugins/org.talend.librariesmanager/src/main/java/org/talend/librariesmanager/model/ModulesNeededProvider.java [hunk 2]
```diff
@@ -317,9 +317,9 @@ public static void resetCurrentJobNeededModuleList(IProcess process) {
                 getAllManagedModules().add(toAdd);
             }
 
-            ILibraryManagerService libManagerService = (ILibraryManagerService) GlobalServiceRegister.getDefault()
-                    .getService(ILibraryManagerService.class);
-            libManagerService.saveCustomMavenURIMap();
+//            ILibraryManagerService libManagerService = (ILibraryManagerService) GlobalServiceRegister.getDefault()
+//                    .getService(ILibraryManagerService.class);
+//            libManagerService.saveCustomMavenURIMap();
```
Comment (nrousseau) [inline main/plugins/org.talend.librariesmanager/src/main/java/org/talend/librariesmanager/model/ModulesNeededProvider.java]: remove code unused

File: main/plugins/org.talend.librariesmanager/src/main/java/org/talend/librariesmanager/model/ModulesNeededProvider.java [hunk 3]
```diff
@@ -306,7 +306,7 @@ public static void resetCurrentJobNeededModuleList(IProcess process) {
                     toAdd = new ModuleNeeded("Job " + process.getName(), "Required for the job " + process.getName() + ".", true,
                             neededLibrary.getMavenUri());
 
-                    toAdd.setCustomMavenUri(neededLibrary.getMavenUri());
+//                    toAdd.setCustomMavenUri(neededLibrary.getMavenUri());
```
Comment (nrousseau) [inline main/plugins/org.talend.librariesmanager/src/main/java/org/talend/librariesmanager/model/ModulesNeededProvider.java]: remove code unused

File: main/plugins/org.talend.librariesmanager.ui/src/main/java/org/talend/librariesmanager/ui/actions/ImportExternalJarAction.java (hunk 1 above)
Comment (jiezhang-tlnd) [reply main/plugins/org.talend.librariesmanager.ui/src/main/java/org/talend/librariesmanager/ui/actions/ImportExternalJarAction.java]: ok, I will remove this unused code.

**Pull request merged**